packages/application-components/src/components/modal-pages/internals/default-form-buttons.tsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import { useIntl, MessageDescriptor } from 'react-intl';
+import PrimaryButton from '@commercetools-uikit/primary-button';
+import SecondaryButton from '@commercetools-uikit/secondary-button';
+import IconButton from '@commercetools-uikit/icon-button';
+import { BinLinearIcon } from '@commercetools-uikit/icons';
+import { sharedMessages } from '@commercetools-frontend/i18n';
+import filterDataAttributes from '../../../utils/filter-data-attributes';
+
+type Label = string | MessageDescriptor;
+type Props = {
+  label: Label;
+  onClick: (event: React.SyntheticEvent) => void;
+  isDisabled: boolean;
+  dataAttributes: { [key: string]: string };
+  children?: never;
+};
+
+const primaryDefaultProps: Pick<
+  Props,
+  'label' | 'isDisabled' | 'dataAttributes'
```

Comment:
Nit: I would not inline these types have have it ontop `type PrimaryDefaultProps = ...` 🤷‍♂  sorry :)